packages/11ty/_includes/components/table-of-contents/grid-item.js

```diff
@@ -0,0 +1,119 @@
+const path = require ('path')
+const { oneLine } = require('common-tags')
+
+/**
+ * Renders a TOC item
+ *
+ * @param     {Object} context
+ * @param     {String} params
+ * @property  {Array} children - The TOC page item's child pages
+ * @property  {String} page - The TOC item's page data
+ *
+ * @return {String} TOC item markup
+ */
+module.exports = function (eleventyConfig) {
+  const contributorList = eleventyConfig.getFilter('contributorList')
+  const getFigure = eleventyConfig.getFilter('getFigure')
+  const getObject = eleventyConfig.getFilter('getObject')
+  const icon = eleventyConfig.getFilter('icon')
+  const markdownify = eleventyConfig.getFilter('markdownify')
+  const pageTitle = eleventyConfig.getFilter('pageTitle')
+  const tableOfContentsImage = eleventyConfig.getFilter('tableOfContentsImage')
+  const urlFilter = eleventyConfig.getFilter('url')
+  const { imageDir } = eleventyConfig.globalData.config.params
+
+  return function (params) {
+    const {
+      children='',
+      classes=[],
+      page
+    } = params
+
+    const {
+      contributor: pageContributors,
+      figure: pageFigure,
+      image,
+      label,
+      layout,
+      object: pageObject,
+      online,
+      short_title,
+      subtitle,
+      summary,
+      title,
+      weight
+    } = page.data
+
+    const isOnline = online !== false
+
+    const pageContributorList = contributorList({ contributors: pageContributors })
+    const pageContributorsElement = pageContributorList
+      ? `<span class="contributor"> — ${pageContributorList}</span>`
+      : ''
+    const pageTitleElement = oneLine`${pageTitle({ label, subtitle, title })}${pageContributorsElement}`
+    const arrowIcon = `<span class="arrow remove-from-epub">&nbsp${icon({ type: 'arrow-forward', description: '' })}</span>`
+    let mainElement = `${markdownify(pageTitleElement)}${isOnline && !children ? arrowIcon : ''}`
+    /**
+     * Create cards for page items
+     */
+    const imageAttribute = image || pageFigure || pageObject ? "image" : "no-image"
+    const slugPageAttribute = children ? "slug-page" : ""
+
+    let imageElement
+    switch (true) {
+      case !!image:
+        imageElement = `<div class="card-image">
+            <figure class="image">
+              <img src="${path.join(imageDir, image)}" alt="" />
+            </figure>
+          </div>`
```

Comment:
Consistently using a `common-tags` method may produce better formatted markup?
```suggestion
        imageElement = html`
          <div class="card-image">
            <figure class="image">
              <img src="${path.join(imageDir, image)}" alt="" />
            </figure>
          </div>
        `
```